breadbox/breadbox/models/dataset.py

```diff
@@ -290,68 +289,52 @@ class DimensionSearchIndex(Base, UUIDMixin, GroupMixin):
     __tablename__ = "dimension_search_index"
 
     __table_args__ = (
-        Index("idx_dimension_search_index_dimension_id", "dimension_id",),
         # the following indices are for speeding up querying to search index, but
         # not 100% confident this is the optimal combination of indices. However,
         # testing the get dimensions endpoint suggests that these are "good enough"
         # for now.
         Index("idx_dim_search_index_perf_1", "value"),
-        Index("idx_dim_search_index_perf_2", "type_name", "value"),
+        Index("idx_dim_search_index_perf_2", "dimension_type_name", "value"),
         Index(
-            "idx_dim_search_index_perf_3", "dimension_given_id", "type_name", "value"
+            "idx_dim_search_index_perf_4",
+            "dimension_given_id",
+            "dimension_type_name",
+            "value",
         ),
     )
 
-    dimension_id = Column(
-        String, ForeignKey("dimension.id", ondelete="CASCADE"), nullable=False
-    )
-    dimension = relationship(
-        Dimension,
-        backref=backref(
-            "dimension_search_indexes",
-            cascade="all, delete-orphan",
-            passive_deletes=True,
-        ),
-    )
-    dimension_given_id = Column(String, nullable=False)
-    label = Column(String, nullable=False)
+    # The dimension which owns the property. (One should be able to group by dimension_id to get the bag of words associated with that dimension)
+    # dimension_id = Column(
+    #     String, ForeignKey("dimension.id", ondelete="CASCADE"), nullable=False
+    # )
+    # dimension = relationship(
+    #     Dimension,
+    #     backref=backref(
+    #         "dimension_search_indexes",
+    #         cascade="all, delete-orphan",
+    #         passive_deletes=True,
+    #     ),
+    # )
     property = Column(String, nullable=False)
     value = Column(String, nullable=True)
-    priority = Column(Integer, nullable=True)
-    axis = Column(String, nullable=False)
-    type_name = Column(String, nullable=False)
-
-
-class DatasetReference(Base, UUIDMixin, GroupMixin, ReferencedGroupMixin):
```

Comment:
Fyi, now that this `DatasetReference` table is gone, you should be able to remove some of the complicated access control workarounds I'd added. Specifically in `breadbox/models/group.py` you can remove the definitions of:
* `ReferencedGroupMixin`
* `_add_filtering_criteria`

This is the only place those workarounds are used currently, and I can't think of anything else that would need them in the future. I believe we've decided that "dimension types" are always public, so there's no more need for the concept of "referenced groups".